test/integration/controller-manager/matching_test.go

```diff
@@ -64,11 +68,30 @@ const NumObjEnvar = "CONTROLLER_TEST_NUM_OBJECTS"
 // YOU MUST HAVE THE ETCD BINARY ON YOUR `$PATH`.
 //
 // This test exercises the workload matching functionality.
-// This test generates a given number of objects randomly.
-// 2/3 of them are created in the apiserver.
-// DownsyncObjectTests are extracted from 2/3 of them, with random tweaks.
-// This test creates a BindingPolicy object with those DownsyncObjectTests.
-// This test then waits up to 1 minute for a Binding with expected matches to appear.
+// The test maintains a set of workload objects, which varies throughout the test.
+// The test maintains a set of DownsyncObjectTests, which varies throughout the test.
+// The DownsyncObjectTests are derived from, on average, 2/3 of the workload objects,
+// chosen uniformly at random. The derived test looks for the base object but with tweaks
+// to make it match more or fewer.
+// An average of 2/3 of the workload objects, chosen randomly, have been created in the apiserver
+// at any given moment; the rest have not been created or have also been deleted.
+//
+// The test proceeds through a series of rounds.
+// There are 7 types of rounds, identified by the numbers 1 through 7.
```

Comment:
Perhaps these types (and other numbers that are used to identify things) can be explicit as consts? It would be easier to know what's going on in the code that way.

Reply:
This type is three bit fields, enumerated constants would be cumbersome and misleading.

Reply:
I introduced three constants for the bit fields.